src/IRI.php

```diff
@@ -539,19 +540,22 @@ protected function remove_iunreserved_percent_encoded($match)
         // at the first byte!).
         $string = '';
         $remaining = 0;
+        $start = 0;
+        $character = 0;
+        $length = 0;
 
         // Loop over each and every byte, and set $value to its value
         for ($i = 1, $len = count($bytes); $i < $len; $i++) {
             $value = hexdec($bytes[$i]);
 
+            // By default we are valid
+            $valid = true;
```

Comment:
While I do not think this changes the semantics, it might be clearer to only reset this when an initial byte is read.

Reply:
done.